include/deal.II/matrix_free/tools.h

```diff
@@ -157,6 +157,184 @@ namespace MatrixFreeTools
     const unsigned int first_selected_component = 0);
 
 
+
+  /**
+   * A wrapper around MatrixFree to help users to deal with DoFHandlers,
+   * where there are cells without degrees of freedom, i.e., with
+   * FENothing assigned to them. In the following we call such cells
+   * deactivated. All other cells are activated. In contrast to MatrixFree,
+   * this class skips deactivated cells and faces between activated and
+   * deactivated cells are treated as boundary faces.
```

Comment:
I think it would be important to say that this class considers only the first `FE` in an `hp::FECollection` that has non-zero number of unknowns (if I read the code correctly).

Reply:
I have added a comment. To lift this restriction is the next step; see https://github.com/dealii/dealii/pull/14121#discussion_r919035525.